pkg/scheduler/algorithm/well_known_labels.go

```diff
@@ -36,6 +36,11 @@ const (
 	// It is deprecated since 1.9
 	DeprecatedTaintNodeUnreachable = "node.alpha.kubernetes.io/unreachable"
 
+	// TaintNodeUnschedulable will be added when node becomes unschedulable
+	// and feature-gate for TaintNodesByCondition flag is enabled,
+	// and removed when node becomes scheduable.
+	TaintNodeUnschedulable = "node.kubernetes.io/unschedulable"
```

Comment:
How are we making sure that no ordinary (other than DS pods) pods is ever allowed to have this toleration to really have the affect of unschedulable, like via admission plugin (podtolerationrestriction)?

Reply:
We actually do not enforce it. In fact, regular pods may add this toleration and it is fine. Probably most users do not want to add this to their pods though, because an node that is not schedulable, is probably not able to run the pod or even if it does, it probably has other issues, such as resource exhaustion, etc.

Reply:
> How are we making sure that no ordinary (other than DS pods) pods is ever allowed to have this toleration to really have the affect of unschedulable

from the system's perspective, there is no such thing as a "DS pod". whatever mechanisms we have must work generically.

pods can already skip scheduling by setting a nodeName directly

Reply:
So sounds like behavior of "node.kubernetes.io/unschedulable" is different from unschedulable. Wonder if it might cause issue during node drains where unscheduable means no pods (perhaps except DS) can be scheduled. But with "node.kubernetes.io/unschedulable" , some regular pods that had the compatible toleration for "node.kubernetes.io/unschedulable" (for whatever reasons, intentionally or erroneously) might still get scheduled.

Reply:
> Wonder if it might cause issue during node drains where unscheduable means no pods can be scheduled

that does seem likely

> pods that had the compatible toleration for "node.kubernetes.io/unschedulable" (for whatever reasons, intentionally or erroneously) might still get scheduled

that doesn't seem concerning... I doubt a pod tolerating a noschedule effect of a node.kubernetes.io/unschedulable taint would be surprised it got scheduled